AK/SIMDExtras.h

```diff
@@ -286,4 +320,40 @@ ALWAYS_INLINE static T elementwise_byte_reverse(T a)
     return Detail::elementwise_byte_reverse_impl(a, MakeIndexSequence<vector_length<T>>());
 }
 
+template<SIMDVector T>
+ALWAYS_INLINE static ElementOf<T> reduce_or(T const& a)
+{
+    static_assert(is_power_of_two(vector_length<T>));
+    static_assert(IsUnsigned<ElementOf<T>>);
+
+#if defined __has_builtin
+#    if __has_builtin(__builtin_reduce_or)
+    if (true) {
+        return __builtin_reduce_or(a);
+    } else
+#    endif
```

Comment:
Weird pattern, shouldnt `if constexpr(...)` work as well and be nicer?

Reply:
Also in which case is `__has_builtin` not defined, we dont really care for MSVC

Reply:
I was trying to fix this problem: https://github.com/SerenityOS/serenity/actions/runs/10498331182/job/29083055403#step:10:2402 it seems that compiler doesn't like the `if constexpr` pattern in that specific case.

Reply:
Ah sure, then go through the pre-processor, but the `#if defined __has_builtin` should be redundant

Reply:
Oh cross that, making it
`__has_builtin(__builtin_reduce_or) && DependentTrue<T>` should make it work
(weird rules around when a false constexpr branch is checked)